### PR TITLE
Implement SelectInput component

### LINKED ISSUE #201: Align select visual styles and interaction patterns across app
## Is your feature request related to a problem? Please describe.
Selects should look and act the same way across the site. They currently do not. The search dropdown has custom styles and uses different HTML elements than the sort results or format dropdowns.

## Describe the solution you'd like
Make sure that selects behave and appear uniformly across the site.

### PR DESCRIPTION
**Pull request by helrond**
Implements a SelectInput component in search bar, search results sort, and duplication modal, which uses downshift's [useSelect](https://github.com/downshift-js/downshift/tree/master/src/hooks/useSelect) hook. Styling has been aligned across all instances of the select. In forms, the selected value is supplied via a hidden input.

There are also some styling tweaks to the confirm modals that are part of this PR, and I made sure that the confirm modal is shown when mobile users click on "delete all items".

fixes #201

### CODE REVIEW

File: src/components/ModalConfirm/index.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const ModalConfirm = props => (
     </div>
     <div className="modal-body">
       <div className="modal-message">
-        <p>{props.message}</p>
```
Comment (HaSistrunk) [inline src/components/ModalConfirm/index.js:23]: Removing the `<p>` here appears to mess up the text formatting in the modal message (see the Request in Reading Room confirmation).

**Comment by HaSistrunk** [comment]
The SelectInput looks good, and I think is a solid improvement. I'm seeing a couple things with this implementation:

1. I think because we're using the `InputLabel` component for the label, the ids aren't lining up correctly. We should have:<label for="downshift-0-label"...` and `<button id="downshift-0-toggle-button"...`

**Comment by HaSistrunk** [comment]
The SelectInput looks good, and I think is a solid improvement. I'm seeing a couple things with this implementation:

1. I think because we're using the `InputLabel` component props with the label, the ids aren't lining up correctly. I believe we should have:
`<label id="downshift-0-label" for="downshift-0-toggle-button"...>` 
`<button id="downshift-0-toggle-button" aria-labelledby="downshift-0-label downshift-0-toggle-button"...>`
`<ul  id="downshift-0-menu" aria-labelledby="downshift-0-label"...>`

2. When I test with a screenreader, once I select something, it reads "undefined has been selected." I haven't been able to figure out why it won't read the value of the thing being selected. I also tested the example implementation in the downshift documentation, and the sr announces what option has been selected just fine in there.

**Comment by helrond** [comment]
Thanks for this close review @HaSistrunk! I believe I have addressed all of the outstanding issue here, along with a bug in the Duplication Modal that was causing the form not to submit (and a typo which prevented an error message from being displayed)

**Comment by HaSistrunk** [comment]
Looks great!

**Review by HaSistrunk — APPROVED** [review]

**Pull request merged**